packages/tab/src/react/list-item.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+  names as themeNames,
+  useTheme
+} from '@pluralsight/ps-design-system-theme'
+import { ValueOf } from '@pluralsight/ps-design-system-util'
+import { css } from 'glamor'
+import React from 'react'
+
+import stylesheet from '../css'
+
+const styles = {
+  bar: () => css(stylesheet['.psds-tab__list-item__bar']),
+  listItem: (active: boolean, themeName: ValueOf<typeof themeNames>) =>
+    css(
+      stylesheet['.psds-tab__list-item'],
+      stylesheet[`.psds-tab__list-item.psds-theme--${themeName}`],
+      active && stylesheet[`.psds-tab__list-item.psds-tab__list-item--active`],
+      active &&
+        stylesheet[
+          `.psds-tab__list-item.psds-tab__list-item--active.psds-theme--${themeName}`
+        ]
+    ),
+  textInner: () => css(stylesheet['.psds-tab__list-item__text-inner']),
+  textWidth: () => css(stylesheet['.psds-tab__list-item__text'])
+}
+
+export interface BaseListItemProps {
+  id: string | number
+  active?: boolean
+  onClick?: (i: number, event: React.MouseEvent) => void
+}
+export type ListItemButtonTagProps = BaseListItemProps &
+  Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'id' | 'onClick'>
+export type ListItemAnchorTagProps = BaseListItemProps &
+  Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'id' | 'onClick'>
+type ListItemProps = ListItemButtonTagProps | ListItemAnchorTagProps
```

Comment:
by using a union and not an overload we lose some type safety. i setup an example here

https://codesandbox.io/s/angry-robinson-woy3s?file=/src/Item.test.tsx

in the example you'll see that we are expecting an anchor tag but the props are not properly typ checked as such

Reply:
Thank you for looking critically at this.  

Please be more specific with this so I can see where we're not getting proper type checking.  Which part is not getting typechecked in the example but is wrong?

I do see this, fwiw:
<img width="1408" alt="Screen Shot 2020-11-10 at 6 00 05 AM" src="https://user-images.githubusercontent.com/552044/98677409-36c60100-231a-11eb-933c-d35512d66c68.png">

Reply:
yes of course. the typing error in the example is actually correct and i'm showing how an `a` tag should be typed. 

the `ListItem` in the example should be an `a` since it has an `href` but it's wrongly accepting an eventHandler with a `HTMLButtonElement` type

Reply:
ah, I finally see it.  The typings do not prevent mixing/matching anchor/button stuff.  And this shows that in practice.  Thanks for the codesandbox and the explanation.  I will play with type overloads.